From 6d24f36c1f89eca9ce249b225ab98360554769ed Mon Sep 17 00:00:00 2001
From: Zhang Jiejing <jiejing.zhang@freescale.com>
Date: Wed, 16 May 2012 11:20:00 +0800
Subject: ENGR00209454 imx6sl: fix build failure and clear warnning message.

fix build failure invoke by reboot function patch,
and refine the code to clear the warnning message.

Signed-off-by: Zhang Jiejing <jiejing.zhang@freescale.com>
---
 arch/arm/plat-mxc/include/mach/system.h |   17 +++++++++++++++++
 arch/arm/plat-mxc/system.c              |   17 ++---------------
 2 files changed, 19 insertions(+), 15 deletions(-)

diff --git a/arch/arm/plat-mxc/include/mach/system.h b/arch/arm/plat-mxc/include/mach/system.h
index 7021c85..c1dfe25 100755
--- a/arch/arm/plat-mxc/include/mach/system.h
+++ b/arch/arm/plat-mxc/include/mach/system.h
@@ -25,4 +25,21 @@ extern void mx5_cpu_lp_set(enum mxc_cpu_pwr_mode mode);
 void arch_idle(void);
 
 void arch_reset(char mode, const char *cmd);
+
+#ifdef CONFIG_MXC_REBOOT_MFGMODE
+void do_switch_mfgmode(void);
+void mxc_clear_mfgmode(void);
+#else
+#define do_switch_mfgmode() do {} while (0)
+#define mxc_clear_mfgmode() do {} while (0)
+#endif
+
+#ifdef CONFIG_MXC_REBOOT_ANDROID_CMD
+void do_switch_recovery(void);
+void do_switch_fastboot(void);
+#else
+#define do_switch_recovery() do {} while (0)
+#define do_switch_fastboot() do {} while (0)
+#endif
+
 #endif /* __ASM_ARCH_MXC_SYSTEM_H__ */
diff --git a/arch/arm/plat-mxc/system.c b/arch/arm/plat-mxc/system.c
index a2830bd..d2a999b 100644
--- a/arch/arm/plat-mxc/system.c
+++ b/arch/arm/plat-mxc/system.c
@@ -21,8 +21,10 @@
 #include <linux/io.h>
 #include <linux/err.h>
 #include <linux/delay.h>
+#include <linux/string.h>
 #include <mach/hardware.h>
 #include <mach/common.h>
+#include <mach/system.h>
 #include <asm/proc-fns.h>
 #include <asm/system.h>
 #ifdef CONFIG_SMP
@@ -32,21 +34,6 @@
 
 static void __iomem *wdog_base;
 
-#ifdef CONFIG_MXC_REBOOT_MFGMODE
-void do_switch_mfgmode(void);
-void mxc_clear_mfgmode(void);
-#else
-void do_switch_mfgmode() {}
-void mxc_clear_mfgmode() {}
-#endif
-
-#ifdef CONFIG_MXC_REBOOT_ANDROID_CMD
-void do_switch_recovery(void);
-void do_switch_fastboot(void);
-#else
-void do_switch_recovery() {}
-void do_switch_fastboot() {}
-#endif
 
 static void arch_reset_special_mode(char mode, const char *cmd)
 {
-- 
1.7.9.5

